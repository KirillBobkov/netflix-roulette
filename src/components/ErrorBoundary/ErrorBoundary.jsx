import React from 'react';
import PropTypes from 'prop-types';
import './ErrorBoundary.scss';

export class ErrorBoundary extends React.Component {
  static defaultProps = {
    children: null
  };

  state = {
      hasError: false,
      message: 'Something went wrong'
  };   

  static getDerivedStateFromError({ message }) {
    return { hasError: true, message };
  }

  renderFallbackUI = () => {
           const {message} = this.state;
           return (
             <div className='error'>
               <span className='error__message'>
                 {message}
               </span>
             </div>
           );
       }

  render() {
     const { hasError } = this.state;
     const { children } = this.props;

     return hasError
          ? this.renderFallbackUI()
          : children;
  };
}

ErrorBoundary.propTypes = {
  children: PropTypes.node
};