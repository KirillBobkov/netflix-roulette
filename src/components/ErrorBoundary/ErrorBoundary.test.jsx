import React from 'react';
import { ErrorBoundary } from './ErrorBoundary';
import { shallow, mount } from 'enzyme';

describe('Error Boundary component', () => {
    
    it('should be render correctly', () => {
        const component = shallow(<ErrorBoundary />);

        expect(component).toMatchSnapshot();
    });

    it('ErrorBoundary not render message without error', () => {
        const component = mount(<ErrorBoundary />);
        const errorElement = component.find('.error');

        expect(component.contains(errorElement)).toEqual(false);
    });

    it('check value "hasError" of ErrorBounary state', () => {
        const component = mount(<ErrorBoundary />);

        expect(component.state('hasError')).toEqual(false);
    });
});