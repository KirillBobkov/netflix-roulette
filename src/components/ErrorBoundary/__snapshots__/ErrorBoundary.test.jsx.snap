// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary component should be render correctly 1`] = `ShallowWrapper {}`;
